plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/functions/tablechanges/TableChangesFunction.java

```diff
@@ -0,0 +1,186 @@
+{
+    private static final String FUNCTION_NAME = "table_changes";
+    private static final String SCHEMA_VAR_NAME = "SCHEMA";
+    private static final String TABLE_VAR_NAME = "TABLE";
+    private static final String START_SNAPSHOT_VAR_NAME = "START_SNAPSHOT_ID";
+    private static final String END_SNAPSHOT_VAR_NAME = "END_SNAPSHOT_ID";
+
+    private final TrinoCatalogFactory trinoCatalogFactory;
+    private final TypeManager typeManager;
+
+    @Inject
+    public TableChangesFunction(TrinoCatalogFactory trinoCatalogFactory, TypeManager typeManager)
+    {
+        super(
+                "system",
+                FUNCTION_NAME,
+                ImmutableList.of(
+                        ScalarArgumentSpecification.builder()
+                                .name(SCHEMA_VAR_NAME)
+                                .type(VarcharType.createUnboundedVarcharType())
+                                .build(),
+                        ScalarArgumentSpecification.builder()
+                                .name(TABLE_VAR_NAME)
+                                .type(VarcharType.createUnboundedVarcharType())
+                                .build(),
+                        ScalarArgumentSpecification.builder()
+                                .name(START_SNAPSHOT_VAR_NAME)
+                                .type(BIGINT)
+                                .build(),
+                        ScalarArgumentSpecification.builder()
+                                .name(END_SNAPSHOT_VAR_NAME)
+                                .type(BIGINT)
+                                .build()),
+                GENERIC_TABLE);
+
+        this.trinoCatalogFactory = requireNonNull(trinoCatalogFactory, "trinoCatalogFactory is null");
+        this.typeManager = requireNonNull(typeManager, "typeManager is null");
+    }
+
+    @Override
+    public TableFunctionAnalysis analyze(ConnectorSession session, ConnectorTransactionHandle transaction, Map<String, Argument> arguments, ConnectorAccessControl accessControl)
+    {
+        String schema = ((Slice) checkNonNull(((ScalarArgument) arguments.get(SCHEMA_VAR_NAME)).getValue())).toStringUtf8();
+        String table = ((Slice) checkNonNull(((ScalarArgument) arguments.get(TABLE_VAR_NAME)).getValue())).toStringUtf8();
+        long startSnapshotId = (long) checkNonNull(((ScalarArgument) arguments.get(START_SNAPSHOT_VAR_NAME)).getValue());
+        long endSnapshotId = (long) checkNonNull(((ScalarArgument) arguments.get(END_SNAPSHOT_VAR_NAME)).getValue());
+
+        SchemaTableName schemaTableName = new SchemaTableName(schema, table);
+        Table icebergTable = trinoCatalogFactory.create(session.getIdentity())
+                .loadTable(session, schemaTableName);
+
+        checkSnapshotExists(icebergTable, startSnapshotId);
+        checkSnapshotExists(icebergTable, endSnapshotId);
+
+        ImmutableList.Builder<Descriptor.Field> columns = ImmutableList.builder();
+        icebergTable.schema().columns().stream()
```

Comment:
Do i understand correctly that this will return the newest schema right ? 
will it be correct to read values from earliest snapshot ?